llvm/lib/Transforms/Vectorize/LoopIdiomVectorize.cpp

```diff
@@ -742,6 +682,11 @@ void AArch64LoopIdiomTransform::transformByteCompare(
   DomTreeUpdater DTU(DT, DomTreeUpdater::UpdateStrategy::Lazy);
   Builder.SetCurrentDebugLocation(PHBranch->getDebugLoc());
 
+  // Safeguard to check if we build the correct DomTree with DTU.
+  auto CheckDTU = llvm::make_scope_exit([&]() {
```

Comment:
Should this be a separate patch?

Reply:
I agree. These DTU tree changes are now reverted.